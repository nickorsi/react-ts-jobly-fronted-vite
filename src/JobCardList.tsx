import React from "react";
import JobCard from "./JobCard";

function JobCardList() {
    return (
        <>
            <div>JobCardList Component</div>
            <JobCard></JobCard>
        </>
    )
}

export default JobCardList;