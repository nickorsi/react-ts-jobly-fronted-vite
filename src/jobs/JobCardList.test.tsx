import {describe, expect, it} from "vitest";
import { render } from "@testing-library/react";

import { JobCardList } from "./JobCardList.tsx";

/**
 * Something wrong with test config, uncomment line below to get intellisense to
 * recognize the extended expect methods, comment out for tests to run properly.
*/
// import "@testing-library/jest-dom";

const TEST_JOBS = [
    {
      id: 1,
      title: "Job 1",
      salary: 1,
      equity: "1",
      companyHandle: "c1",
      companyName: "Company 1",
    },
    {
      id: 2,
      title: "Job 2",
      salary: 2,
      equity: "2",
      companyHandle: "c2",
      companyName: "Company 2",
    },
    {
      id: 3,
      title: "Job 3",
      salary: 3,
      equity: "3",
      companyHandle: "c3",
      companyName: "Company 3",
    },
  ];


describe("Simple Render Tests", () => {
    it("renders without crashing", function () {
        render(<JobCardList jobs={TEST_JOBS}/>);
    });
    it("matches the snapshot", function () {
        const { container } = render(<JobCardList jobs={TEST_JOBS}/>);

        expect(container).toMatchSnapshot();
    });
    it("renders with expected info", function () {
        const { container } = render(<JobCardList jobs={TEST_JOBS}/>);

        expect(container).toContainHTML("Job 1");
        expect(container).toContainHTML("Job 2");
        expect(container).toContainHTML("Job 3");
    });
});