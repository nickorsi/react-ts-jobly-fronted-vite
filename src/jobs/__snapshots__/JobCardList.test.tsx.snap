// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Simple Render Tests > matches the snapshot 1`] = `
<div>
  <div
    class="job card"
    id="1"
  >
    <div
      class="job-titles"
    >
      <p>
        Job Title: 
        <span
          class="job-title"
        >
          Job 1
        </span>
      </p>
      <p
        class="job-company"
      >
        Company: 
        Company 1
      </p>
    </div>
    <div
      class="job-pay-info"
    >
      <p
        class="job-salary"
      >
        Salary: 
        1
      </p>
      <p
        class="job-equity"
      >
        Equity: 
        1
      </p>
    </div>
  </div>
  <div
    class="job card"
    id="2"
  >
    <div
      class="job-titles"
    >
      <p>
        Job Title: 
        <span
          class="job-title"
        >
          Job 2
        </span>
      </p>
      <p
        class="job-company"
      >
        Company: 
        Company 2
      </p>
    </div>
    <div
      class="job-pay-info"
    >
      <p
        class="job-salary"
      >
        Salary: 
        2
      </p>
      <p
        class="job-equity"
      >
        Equity: 
        2
      </p>
    </div>
  </div>
  <div
    class="job card"
    id="3"
  >
    <div
      class="job-titles"
    >
      <p>
        Job Title: 
        <span
          class="job-title"
        >
          Job 3
        </span>
      </p>
      <p
        class="job-company"
      >
        Company: 
        Company 3
      </p>
    </div>
    <div
      class="job-pay-info"
    >
      <p
        class="job-salary"
      >
        Salary: 
        3
      </p>
      <p
        class="job-equity"
      >
        Equity: 
        3
      </p>
    </div>
  </div>
</div>
`;
